keps/sig-scheduling/4832-async-preemption/README.md

```diff
@@ -0,0 +1,415 @@
+having feature tests is enough.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The partly failure in the rollout isn't there because the scheduler is only the component to rollout this feature. 
+But, if upgrading the scheduler itself fails somehow, new Pods won't be scheduled anymore. 
+If there's a bug in the preemption because of this enhancement, and also downgrading the scheduler fails somehow,
+running Pods could be affected, for example, by being deleted by mistake (depending on bugs).
+
+###### What specific metrics should inform a rollback?
+
+Maybe something goes wrong with the preemption if `goroutines_duration_seconds{operation=preemption}` takes too long time.
+Also, if `preemption_attempts_total` increases too much, then that might also imply some bugs around the preemption.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No. This feature is an in-memory feature of the scheduler, 
+and just upgrading it and upgrade->downgrade->upgrade are both the same.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+This feature is used during all Pods' preemption if the feature gate is enabled.
+You can find Pods that have triggered the preemption by referring to `.Status.NominatedNodeName`,
+and Pods that have been preempted by referring to their condition with `type: DisruptionTarget` and `reason: PreemptionByScheduler`.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] API .status
+  - Other field: If `.Status.NominatedNodeName` of Pods is non-empty, they have experienced the preemption running asynchronously.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+- The failure rate of the preemption goroutine (`goroutines_execution_total{result=error, operation=preemption}`/`goroutines_execution_total{operation=preemption}`) should be < 0.01.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: `goroutines_execution_total{result=error, operation=preemption}` 
+  - Components exposing the metric: kube-scheduler
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+- `goroutines_duration_seconds` (w/ label: `operation`): to observe how many preemption goroutines have failed.
+- `goroutines_execution_total` (w/ labels: `operation`, `result`): to observe how long each preemption goroutine takes to complete.
```

Comment:
I'm not sure what is the difference between the two. Can you review?

Reply:
Oops. Sorry, the explanation was the opposite.